src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -90,19 +88,32 @@ public static IHostBuilder AddWebScriptHost(this IHostBuilder builder, IServiceP
                 .ConfigureLogging(loggingBuilder =>
                 {
                     loggingBuilder.Services.AddSingleton<ILoggerFactory, ScriptLoggerFactory>();
-
                     loggingBuilder.AddWebJobsSystem<SystemLoggerProvider>();
+
                     if (environment.IsAzureMonitorEnabled())
                     {
                         if (environment.IsConsumptionOnLegion())
                         {
                             loggingBuilder.Services.AddOptions<LoggerFilterOptions>()
                                 .Configure<IOptionsMonitor<AppServiceOptions>>((filters, options) =>
                                 {
-                                    filters.AddFilter<AzureMonitorDiagnosticLoggerProvider>((category, level) =>
+                                    if (options.CurrentValue.IsAzureMonitorLoggingEnabled)
+                                    {
+                                        // Remove any existing filters for AzureMonitorDiagnosticLoggerProvider that may have been added.
+                                        var providerTypeName = typeof(AzureMonitorDiagnosticLoggerProvider).FullName;
+
+                                        foreach (var rule in filters.Rules
+                                            .Where(r => r is { ProviderName: var p, LogLevel: LogLevel.None, CategoryName: null } && p == providerTypeName)
+                                            .ToList())
+                                        {
+                                            filters.Rules.Remove(rule);
+                                        }
```

Comment:
what rules would we be removing here? Are you expecting the rule from like 115 to be present?